Older systems (< 10.6) don't have the kernel support for pthread_threadid_np.
But they do have pthread_mach_thread_np, which is a similar function, if not quite
as ideal in scope. kencu@macports.org

diff --git Python/thread_pthread.h Python/thread_pthread.h
index 5678b05..5cfdd17 100644
--- Python/thread_pthread.h
+++ Python/thread_pthread.h
@@ -325,9 +327,12 @@ PyThread_get_thread_native_id(void)
 {
     if (!initialized)
         PyThread_init_thread();
-#ifdef __APPLE__
+#if defined(__APPLE__) && MACOSX_VERSION_MAX_ALLOWED >= 1060
     uint64_t native_id;
     (void) pthread_threadid_np(NULL, &native_id);
+#elif defined(__APPLE__) && MACOSX_VERSION_MAX_ALLOWED < 1060
+    uint64_t native_id;
+    native_id = pthread_mach_thread_np(pthread_self());
 #elif defined(__linux__)
     pid_t native_id;
     native_id = syscall(SYS_gettid);
